server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/DelayedAllocationIT.java

```diff
@@ -191,7 +190,6 @@ private void indexRandomData() throws Exception {
     private String findNodeWithShard() {
         ClusterState state = client().admin().cluster().prepareState().get().getState();
         List<ShardRouting> startedShards = RoutingNodesHelper.shardsWithState(state.getRoutingNodes(), ShardRoutingState.STARTED);
-        Collections.shuffle(startedShards, random());
-        return state.nodes().get(startedShards.get(0).currentNodeId()).getName();
+        return state.nodes().get(randomFrom(startedShards).currentNodeId()).getName();
```

Comment:
No need to shuffle entire collection to get a single random element.

Reply:
:+1: even better